synapse/tests/test_lib_node.py

```diff
@@ -0,0 +1,168 @@
+
+                self.raises(s_exc.NoSuchProp, node.get, 'neat::tick')  # implicit pivot from neat (not a prop) to tick
+                self.raises(s_exc.NoSuchForm, node.get, 'tick::tick')  # implicit pivot from neat to tick (not a form)
+                self.none(node.get('bar::bar'))  # implicit piviot from bar to bar
+
+    def test_pop(self):
+        raise NotImplementedError()
+
+    def test_repr(self):
+        with self.getTestCore() as core:
+            with core.snap(write=True) as snap:
+
+                form = 'teststr'
+                valu = 'cool'
+                props = {'tick': 12345}
+                node = snap.addNode(form, valu, props=props)
+                self.none(node.repr())
+                self.eq(node.repr('tick'), '1970/01/01 00:00:12.345')
+
+                form = 'testthreetype'
+                valu = 'cool'
+                node = snap.addNode(form, valu)
+                self.eq(node.repr(), '3')
+                reprs = {k: v for (k, v) in node.reprs().items() if not k.startswith('.')}
+                self.eq(reprs.get('three'), '3')
+
+    def test_tags(self):
+        form = 'teststr'
+        valu = 'cool'
+        props = {'tick': 12345}
+
+        with self.getTestCore() as core:
+            with core.snap(write=True) as snap:
+                self.true(snap.strict)
+
+                node = snap.addNode(form, valu, props=props)
+
+                # Add a tag
+                node.addTag('cool', valu=(1, 2))
+                self.eq(node.getTag('cool'), (1, 2))
+                node.addTag('cool', valu=(1, 2))  # Add again
+                self.eq(node.getTag('cool'), (1, 2))
+                node.addTag('cool', valu=(1, 3))  # Add again with different valu
+                self.eq(node.getTag('cool'), (1, 3))
+                node.addTag('cool', valu=(-5, 0))  # Add again with different valu
+                self.eq(node.getTag('cool'), (-5, 0))  # Does not merge, overwrites
+
+                self.true(node.hasTag('cool'))
+                self.true(node.hasTag('#cool'))
+                self.false(node.hasTag('notcool'))
+                self.false(node.hasTag('#notcool'))
+
+                # Demonstrate that valu is only applied at the level that addTag is called
+                node.addTag('cool.beans.abc', valu=(1, 8))
+                self.eq(node.getTag('cool.beans.abc'), (1, 8))
+                self.eq(node.getTag('cool.beans'), (None, None))
+                self.eq(node.getTag('cool'), (-5, 0))  # from above
+
+                self.raises(s_exc.NoSuchProp, node.pop, 'nope')
+                snap.strict = False
+                self.false(node.pop('nope'))
+                snap.strict = True
+
+                with self.getTestDir() as dirn:
+                    with s_auth.Auth(dirn) as auth:
+                        user = auth.addUser('hatguy')
+                        snap.setUser(user)
+
+                        self.raises(s_exc.AuthDeny, node.set, 'tick', 1)
```

Comment:
@invisig0th - Should this raise AuthDeny? It currently does. The user has no perms configured

Reply:
Fixed